server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
@rbarlow I think Pulp has more than just the Celery results backend bug to work properly with mongoDB replicate sets. We have the auto-retry issue and that failover has been untested. You could say it's our [number 1 issue](https://pulp.plan.io/issues/1).

+1 to not using `safe` anywhere in the codebase.

I agree we likely will need to compute write_concern, but do we have to do it now for this specific bugfix? To avoid getting into all that now, couldn't we set write_concern as `write_concern={fsync: True}` and leave off the `w` option? That would ensure the write occurred to the mongodb Pulp is connected to and leaves the rest to be resolved as [issue 1](https://pulp.plan.io/issues/1).